From 8e6d5ae271339908a69c14fb2ea46766ac7500c7 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Tue, 1 Jul 2014 14:50:40 +0100
Subject: [PATCH 239/351] mfd: arizona: Set sane headphone impedance default

The current default for the headphone impedance field is zero ohms which
means that if no extcon driver is built in the headphones will not be
enabled due to short. Provide a sane default for this field.

Change-Id: I03cebf859807f0b606735af435c15c6923c0dff9
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/mfd/arizona-core.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/mfd/arizona-core.c b/drivers/mfd/arizona-core.c
index 493c932..df23310 100644
--- a/drivers/mfd/arizona-core.c
+++ b/drivers/mfd/arizona-core.c
@@ -1465,6 +1465,12 @@ int arizona_dev_init(struct arizona *arizona)
 	arizona_request_irq(arizona, ARIZONA_IRQ_UNDERCLOCKED, "Underclocked",
 			    arizona_underclocked, arizona);
 
+	/**
+	 * Give us a sane default for the headphone impedance in case the
+	 * extcon driver is not used
+	 */
+	arizona->hp_impedance = 32;
+
 	switch (arizona->type) {
 	case WM5102:
 		ret = mfd_add_devices(arizona->dev, -1, wm5102_devs,
-- 
1.7.9.5

